graphsync/main.go

```diff
@@ -0,0 +1,340 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	dss "github.com/ipfs/go-datastore/sync"
+	"github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipfs/go-ipfs-chunker"
+	"github.com/ipfs/go-ipfs-exchange-offline"
+	"github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/go-unixfs/importer/balanced"
+	ihelper "github.com/ipfs/go-unixfs/importer/helpers"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
+	"github.com/testground/sdk-go/network"
+
+	gs "github.com/ipfs/go-graphsync"
+	gsi "github.com/ipfs/go-graphsync/impl"
+	gsnet "github.com/ipfs/go-graphsync/network"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-noise"
+	"github.com/libp2p/go-libp2p-secio"
+	tls "github.com/libp2p/go-libp2p-tls"
+
+	"github.com/testground/sdk-go/run"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+)
+
+var testcases = map[string]interface{}{
+	"stress": run.InitializedTestCaseFn(runStress),
+}
+
+func main() {
+	run.InvokeMap(testcases)
+}
+
+func runStress(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
+	var (
+		size       = runenv.IntParam("size")
+		bandwidths = runenv.SizeArrayParam("bandwidths")
+		latencies  []time.Duration
+	)
+
+	lats := runenv.StringArrayParam("latencies")
+	for _, l := range lats {
+		d, err := time.ParseDuration(l)
+		if err != nil {
+			return err
+		}
+		latencies = append(latencies, d)
+	}
+
+	runenv.RecordMessage("started test instance")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+
+	initCtx.MustWaitAllInstancesInitialized(ctx)
+	defer initCtx.SyncClient.MustSignalAndWait(ctx, "done", runenv.TestInstanceCount)
+
+	host, peers := makeHost(ctx, runenv, initCtx)
+	defer host.Close()
+
+	var (
+		// make datastore, blockstore, dag service, graphsync
+		ds     = dss.MutexWrap(datastore.NewMapDatastore())
+		bs     = blockstore.NewBlockstore(ds)
+		dagsrv = merkledag.NewDAGService(blockservice.New(bs, offline.Exchange(bs)))
+		gsync  = gsi.New(ctx, gsnet.NewFromLibp2pHost(host), makeLoader(bs), makeStorer(bs))
```

Comment:
you can get rid of the makeLoader and makeStorer functions and just use these utility functions here that do the exact same thing (I should have removed them from the integration tests to not create confusion)

https://github.com/ipfs/go-graphsync/blob/master/storeutil/storeutil.go#L16
https://github.com/ipfs/go-graphsync/blob/master/storeutil/storeutil.go#L32

Reply:
Done, thanks!